backends/mediatek/neuropilot/quantizer/annotator.py

```diff
@@ -0,0 +1,263 @@
+            torch.ops.aten.mul.Tensor,
+            torch.ops.aten.mul_.Scalar,
+            torch.ops.aten.mul_.Tensor,
+            torch.ops.aten.rsub.Scalar,
+            torch.ops.aten.rsub.Tensor,
+            torch.ops.aten.sub.Scalar,
+            torch.ops.aten.sub.Tensor,
+            torch.ops.aten.sub_.Scalar,
+            torch.ops.aten.sub_.Tensor,
+        ]:
+            _annotate_output_qspec(relu_node, quant_config.activation)
+            _mark_as_annotated([producer_node, relu_node])
+            continue
+
+
+def _annotate_rmsnorm_pattern(graph: Graph, quant_config: QuantizationConfig) -> None:
+
+    class ExecuTorchPattern(torch.nn.Module):
+        def forward(self, x):
+            norm = x * torch.rsqrt((x * x).mean(-1, keepdim=True) + 1e-6)
+            return norm, dict()
+
+    class MTKPattern(torch.nn.Module):
+        def forward(self, x):
+            norm = x * torch.rsqrt(x.pow(2).mean(-1, keepdim=True) + 1e-6)
+            return norm, dict()
+
+    for pattern_cls in (ExecuTorchPattern, MTKPattern):
+        pattern_gm = capture_pre_autograd_graph(pattern_cls(), (torch.randn(3, 3),))
+        matcher = SubgraphMatcherWithNameNodeMap(pattern_gm, ignore_literals=True, remove_overlapping_matches=False)
+        matches = matcher.match(graph)
+        for match in matches:
+            target_nodes = []
+            for node in match.nodes_map.values():
+                if node in match.placeholder_nodes:
+                    continue
+                if node.op == 'call_function' and node.target in OP_TO_ANNOTATOR:
+                    target_nodes.append(node)
+
+            if any(_is_annotated(node) for node in target_nodes):
+                continue
+            _mark_as_annotated(target_nodes)
+            for node in match.returning_nodes:
+                _annotate_output_qspec(node, quant_config.activation)
+
+
+def annotate_placeholder(node: Node, quant_config: QuantizationConfig) -> None:
+    if _is_annotated(node):
+        return
+
+    if _is_float_activation_tensor(node):
+        _annotate_output_qspec(node, quant_config.activation)
+
+    _mark_as_annotated([node])
+
+
+@register_annotator(
+    [
+        torch.ops.aten.conv1d.default,
+        torch.ops.aten.conv2d.default,
+        torch.ops.aten.linear.default,
+    ]
+)
+def annotate_affine_ops(node: Node, quant_config: QuantizationConfig) -> None:
+    if _is_annotated(node):
+        return
+
+    weight_node = node.args[1]
+    _annotate_input_qspec_map(
+        node,
+        weight_node,
+        quant_config.weight,
+    )
+    _annotate_output_qspec(node, quant_config.activation)
+
+    # Make weight as annotated because it is a constant node
+    _mark_as_annotated([node, weight_node])
+
+
+@register_annotator(
+    [
+        torch.ops.aten.add.Scalar,
+        torch.ops.aten.add.Tensor,
+        torch.ops.aten.add_.Scalar,
+        torch.ops.aten.add_.Tensor,
+        torch.ops.aten.bmm.default,
```

Comment:
I asked this question in the meeting and actually this part of code answered my question. The quantizer looks reasonable to me now.

Reply:
Yeah, it's more like describing how the ops are quantized.